### PR TITLE
sync with PMIX master

### PR DESCRIPTION
**Pull request by annu13**

### CODE REVIEW

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by annu13** [comment]
Bot: retest

File: opal/mca/pmix/pmix1xx/pmix/examples/server.c [hunk 1]
```diff
@@ -343,6 +346,20 @@ static void errhandler(pmix_status_t status,
...
+                      void *cbdata)
+{
+    pmix_output(0, "SERVER: OP CALLBACK CALLED WITH STATUS %d", status);
+}
+
+static void errhandler_reg_callbk (pmix_status_t status,
+                                   int errhandler_ref,
+                                   void *cbdata)
+{
+    pmix_output(0, "SERVER: ERRHANDLER REGISTRATION CALLBACK CALLED WITH STATUS %d, ref=%d",
```
Comment (rhc54) [inline opal/mca/pmix/pmix1xx/pmix/examples/server.c]: We should probably remove this debug, or at least replace it with pmix_output_verbose

Comment (annu13) [inline opal/mca/pmix/pmix1xx/pmix/examples/server.c]: Will replace with pmix_output_verbose.

**Comment by lanl-ompi** [comment]
Test FAILed.

**Comment by annu13** [comment]
@rhc54, please merge this PR. We need to replace pmix_output with pmix_output_verbose in multiple places, I will clean those and submit a PR in pmix repo directly.

**Comment by jsquyres** [comment]
@rhc54 is probably going to be out for a few days.

Is there a reason you can't update this PR to be correct before we merge it?

**Comment by annu13** [comment]
It is a trivial change that needs to be done in many places in this file and other files in PMIX. I thought it was easier to do it there and bring to OMPI than the other way round. Anyways I just made the changes and added it to the PR. This is good to be merged once the build is successful.

**Comment by jsquyres** [comment]
@annu13 I'm sorry; that's what I actually meant: that you would put this stuff on the PMIX master, and then add another "sync" commit over here on this PR.  I agree that the direction of code should always flow from PMIX to OMPI -- not the other way around.

Have you committed this stuff on the PMIX master yet?

Also, what's the merge commit on this PR?  It doesn't seem like that should be here -- i.e., it seems like that is a problem...?

**Comment by annu13** [comment]
On 10/6/15, 7:30 AM, "Jeff Squyres" <notifications@github.com> wrote:

>@annu13 <https://github.com/annu13> I'm sorry; that's what I actually
>meant: that you would put this stuff on the PMIX master, and then add
>another "sync" commit over here on this PR. I agree that the direction of
>code should
> always flow from PMIX to OMPI -- not the other way around.
>Have you committed this stuff on the PMIX master yet?
The debug print fix commit
(https://github.com/openmpi/ompi/commit/5787e9248f3c720acd5487912936524f16b
37db9) is not in PMIX master. We will fix it in PMIX master bring it in
when we sync next time.
>Also, what's the merge commit on this PR? It doesn't seem like that
>should be here -- i.e., it seems like that is a problem...?
>�
The merge commit corresponds to syncing my fork(annu13/ompi) with
open-mpi/ompi. All the commits were picked up by git automatically. I am
not sure why that commit is causing a failure.
Currently pmix2 is broken on OMPI master, this PR bring the pmix2 fixes
from PMIX master. Could you merge the PR partially ignoring today�s
commits so we can have pmix2 working again.

**Comment by jsquyres** [comment]
@annu13 I'm confused -- I thought you wanted all code to flow from the PMIX tree to OMPI (and not the other way around).  Can you commit the 5787e9248f3c720acd5487912936524f16b patch to PMIX and then re-sync over here to OMPI again?  That might also get rid of the (seemingly) unnecessary merge commit...?

**Comment by annu13** [comment]
fixed debug stmts in pmix master and synced with pmix master (pmix git repo rev git69c398e)

**Pull request merged**